src/components/admin/lib/article-validators.js

```diff
@@ -0,0 +1,21 @@
+export function hasNonHttpsURL(html) {
+  // if (/http:\/\/[\S]+\.[a-zA-Z]{2,}/.test(html) || /http(?!s)/.test(html)) {
+  if (/<a.*?href\s*?=\s*?["'](http(?!s))(.*?)["'] *?>/.test(html)) {
+    return true;
+  } else if (/<img.*?src\s*?=\s*?["'](http)(?!s)(.*?)["'] .*?>/.test(html)) {
```

Comment:
Also here no reason for the capture group around `http` and `.*?` (you do need them around the negative lookahead though! (`(?!s)` as there it isn't a capture group it's part of the negative lookahead syntax)